src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -833,6 +849,46 @@ public static ComponentResultEntity toEntity(LoadFlowResult.ComponentResult comp
                 .bodyToMono(String.class);
     }
 
+    public Mono<String> getSecurityAnalysisStatus(String studyName, String userId) {
+        Objects.requireNonNull(studyName);
+        Objects.requireNonNull(userId);
+
+        return studyRepository.findStudy(userId, studyName).flatMap(entity -> {
+            UUID resultUuid = entity.getSecurityAnalysisResultUuid();
+            return Mono.justOrEmpty(resultUuid).flatMap(uuid -> {
+                String path = UriComponentsBuilder.fromPath(DELIMITER + SECURITY_ANALYSIS_API_VERSION + "/status/{resultUuid}")
+                        .buildAndExpand(resultUuid)
+                        .toUriString();
+                return webClient
+                        .get()
+                        .uri(securityAnalysisServerBaseUri + path)
+                        .retrieve()
+                        .onStatus(httpStatus -> httpStatus == HttpStatus.NOT_FOUND, clientResponse -> Mono.error(new StudyException(SECURITY_ANALYSIS_NOT_FOUND)))
+                        .bodyToMono(String.class);
+            });
+        });
+    }
+
+    public Mono<Void> setSecurityAnalysisStatus(String studyName, String userId, SecurityAnalysisStatus status) {
+        Objects.requireNonNull(studyName);
+        Objects.requireNonNull(userId);
+
+        return studyRepository.findStudy(userId, studyName).flatMap(entity -> {
+            UUID resultUuid = entity.getSecurityAnalysisResultUuid();
```

Comment:
same:



when can resultUuid be empty ? warning: flatmap doesn't execute in this case

Shouldn't it error when the uuid is null, instead of returning an empty mono ?